kevm-pyk/src/kevm_pyk/__main__.py

```diff
@@ -280,6 +280,9 @@ def exec_prove(
             smt_timeout=smt_timeout,
             smt_retry_limit=smt_retry_limit,
         ) as kcfg_explore:
+            for _claim, cfg in proof_problems.items():
+                init_node = cfg.get_unique_init()
+                cfg.replace_node(init_node.id, kcfg_explore.cterm_assume_defined(init_node.cterm))
             proof_problems = {claim: kcfg_explore.simplify(claim, cfg) for claim, cfg in proof_problems.items()}
```

Comment:
Did you verify that this has the intended effect? I guess I'm not sure about Python's modify in place semantics, so I would do something more like:

```suggestion
        ) as kcfg_explore:
            _proof_problems = {}
            for _claim, cfg in proof_problems.items():
                init_node = cfg.get_unique_init()
                cfg.replace_node(init_node.id, kcfg_explore.cterm_assume_defined(init_node.cterm))
                _proof_problems[_claim] = cfg
            proof_problems = _proof_problems
```

@tothtamas28 will the original also modify the `proof_problems` in place?

Also, @dkcumming , we want to do this assuming of definedness _even if_ the user does pass `--no-simplify-init`, which currently it will not enter this block if the user has passed that flag. So we want to move the `if simplify_init: ...`` guard into here.

Reply:
@ehildenb 
From what I could tell with the proofs I did it was working in place, but clearer/safer with the ammedments. 
@tothtamas28 
True, I will refactor.